exporter/opentelemetry-exporter-otlp-proto-http/src/opentelemetry/exporter/otlp/proto/http/_log_exporter/encoder/__init__.py

```diff
@@ -0,0 +1,102 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Sequence, List
+
+from opentelemetry.proto.collector.logs.v1.logs_service_pb2 import (
+    ExportLogsServiceRequest,
+)
+from opentelemetry.proto.logs.v1.logs_pb2 import (
+    ScopeLogs,
+    ResourceLogs,
+)
+from opentelemetry.proto.logs.v1.logs_pb2 import LogRecord as PB2LogRecord
+from opentelemetry.exporter.otlp.proto.http.trace_exporter.encoder import (
+    _encode_instrumentation_scope,
+    _encode_resource,
+    _encode_span_id,
+    _encode_trace_id,
+    _encode_value,
+    _encode_attributes,
+)
+
+
+from opentelemetry.sdk._logs.export import LogData
+
+
+class _ProtobufEncoder:
+    _CONTENT_TYPE = "application/x-protobuf"
+
+    @classmethod
+    def serialize(cls, batch: Sequence[LogData]) -> str:
+        return cls.encode(batch).SerializeToString()
+
+    @staticmethod
+    def encode(batch: Sequence[LogData]) -> ExportLogsServiceRequest:
+        return ExportLogsServiceRequest(
+            resource_logs=_encode_resource_logs(batch)
+        )
+
+
+def _encode_log(log_data: LogData) -> PB2LogRecord:
+    kwargs = {}
+    kwargs["time_unix_nano"] = log_data.log_record.timestamp
+    kwargs["span_id"] = _encode_span_id(log_data.log_record.span_id)
+    kwargs["trace_id"] = _encode_trace_id(log_data.log_record.trace_id)
+    kwargs["flags"] = int(log_data.log_record.trace_flags)
+    kwargs["body"] = _encode_value(log_data.log_record.body)
+    kwargs["severity_text"] = log_data.log_record.severity_text
+    kwargs["attributes"] = _encode_attributes(log_data.log_record.attributes)
+    kwargs["severity_number"] = log_data.log_record.severity_number.value
+
+    return PB2LogRecord(**kwargs)
+
+
+def _encode_resource_logs(batch: Sequence[LogData]) -> List[ResourceLogs]:
+
+    sdk_resource_logs = {}
+
+    for sdk_log in batch:
```

Comment:
Curious if any of the encoding logic can be abstracted out since tracing requires similar logic? Could be left for a different PR though.